lib/ddtrace/profiling/scheduler.rb

```diff
@@ -0,0 +1,66 @@
+require 'ddtrace/worker'
+require 'ddtrace/workers/polling'
+
+module Datadog
+  module Profiling
+    # Pulls and exports profiling data on an async interval basis
+    class Scheduler < Worker
+      include Workers::Polling
+
+      attr_reader \
+        :exporters,
+        :recorder
+
+      def initialize(recorder, exporters, options = {})
+        @recorder = recorder
+        @exporters = [exporters].flatten
+
+        # Workers::Async::Thread settings
+        # Restart in forks by default
+        self.fork_policy = options.fetch(:fork_policy, Workers::Async::Thread::FORK_POLICY_RESTART)
```

Comment:
In the event the application forks, the worker thread will die. `Async::Thread` includes a check to auto-restart the worker thread if its given this "restart" policy. However, the child process must invoke `perform` against the `Scheduler` for this check/restart to occur. We should make sure we call this when the process forks.

Reply:
That's great. I have to tweak that manually still in Python 😓